src/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,284 @@
+            new KeyDownloader(new SignatureFileDownloader($downloader, $this->output)),
+            $untrustedKeyStrategy
+        );
+    }
+
+    private function getCachePath(): string
+    {
+        $cachePath = $this->input->getOption('cache');
+        assert(is_string($cachePath));
+        $this->createDirectory($cachePath);
+
+        if ($this->output->isVeryVerbose()) {
+            $this->output->writeln('Using CACHE: ' . $cachePath);
+        }
+
+        return $cachePath;
+    }
+
+    private function getBaseUri(): string
+    {
+        $baseUri = $this->input->getOption('base-uri');
+        assert(is_string($baseUri));
+
+        $trunk = $this->input->getOption('trunk');
+        assert(is_string($trunk));
+
+        return $baseUri . '/' . $trunk;
+    }
+
+    private function getInstalledRelease(): Release
+    {
+        $application = $this->getApplication();
+        if (null === $application) {
+            throw new RuntimeException('Application is not available');
+        }
+
+        return Release::fromString($application->getVersion());
+    }
+
+    private function shouldUpdate(Release $installedRelease, Release $availableRelease): bool
+    {
+        if ($this->input->getOption('force')) {
+            return true;
+        }
+
+        if ($installedRelease->equals($availableRelease)) {
+            $this->output->writeln('Already version "' . $installedRelease->getVersion() . '" installed');
+
+            return false;
+        }
+
+        $versionParser    = new VersionParser();
+        $installedVersion = $versionParser->normalize($installedRelease->getVersion());
+        $availableVersion = $versionParser->normalize($availableRelease->getVersion());
+
+        if (
+            strnatcmp($installedVersion, $availableVersion) !== 0
+            || $installedRelease->getBuildDate() <= $availableRelease->getBuildDate()
+        ) {
+            return false;
+        }
+
+        if ($this->input->getOption('no-interaction')) {
+            $this->output->writeln(
+                'Installed version is older than available version. Use <info>--force</info> option or use interaction.'
+            );
+            return false;
+        }
+
+        $helper = $this->getHelper('question');
+        assert($helper instanceof QuestionHelper);
+
+        $answer = $helper->ask(
+            $this->input,
+            $this->output,
+            new ConfirmationQuestion(
+                sprintf(
+                    'Installed version has newer built date <info>%s</info> than available <info>%s</info>.'
+                    . ' Really update <info>(y/n)</info>? ',
+                    $installedRelease->getBuildDate()->format('Y-m-d H:i:s T'),
+                    $availableRelease->getBuildDate()->format('Y-m-d H:i:s T')
+                ),
+                false
+            )
+        );
+        assert(is_bool($answer));
+
+        if (! $answer) {
+            return false;
+        }
```

Comment:
Do we really want this that explicit?
I mean, we already have `--force` to ensure a download is always performed and `--dry-run` to ensure no download is performed (nothing is destroyed).
What benefit do we get by adding this question here?

I'd expect an update without questions (as composer does).